Libraries/libutil/Sources/DefaultFilesystem.cpp

```diff
@@ -178,6 +145,55 @@ write(std::vector<uint8_t> const &contents, std::string const &path)
     return true;
 }
 
+bool DefaultFilesystem::
+copyFile(std::string const &from, std::string const &to)
```

Comment:
Several things:

1: Why block off some of the checks, like this->isFile, in the ifdef?  Seems valuable regardless.
2: copyFile() has a fallback implementation?  Why not write this in there?
3: It doesn't look like you're using anything particularly OS-specific.  What is Linux missing prevents you from using the logic in the #ifdef on it?